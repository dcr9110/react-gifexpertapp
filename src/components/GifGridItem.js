import React from 'react';
//impt
import PropTypes from 'prop-types';

export const GifGridItem = ( {title, url} ) => {

    return (
        <div className="card animate__animated animate__jello">
            <p>{ title } </p>
            <img src={ url } alt={ title } />           
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}