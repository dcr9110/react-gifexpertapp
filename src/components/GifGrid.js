import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {
    const { data:images, loading } = useFetchGifs( category );  

    return (
        <>
            <h3> { category }</h3>
              {loading && <h2 className='animate__animated animate__backInDown'>Cargando...</h2> }
            <div className="card-grid">
                {
                    images.map( img =>(
                        <GifGridItem 
                            key={ img.id }
                            { ...img }
                        />
                    ))
                }
            </div>
         </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}