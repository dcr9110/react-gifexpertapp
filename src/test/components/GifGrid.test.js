import { shallow } from 'enzyme';
import React from 'react';
import '@testing-library/jest-dom';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas con el componente <GifGrid/>', () => {
    
    const category = 'samurai';
    
    test('Tiene que mostrar correctamente el componente', () => {
    
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        
        const wrapper = shallow( <GifGrid category={category}/> );
        expect( wrapper ).toMatchSnapshot();
    
    })

    test('debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {
        
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost.jpg',
            title: 'cualquier cosa'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category} />);
        expect( wrapper ).toMatchSnapshot();
        //revisa que el componente no contenga el elemento de cargadon
        expect( wrapper.find('h2').exists() ).toBe( false );
        //revisa el tama√±o del componente
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );

    })
    
    
})
