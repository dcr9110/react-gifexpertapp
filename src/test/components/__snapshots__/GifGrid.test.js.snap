// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con el componente <GifGrid/> Tiene que mostrar correctamente el componente 1`] = `
<Fragment>
  <h3>
     
    samurai
  </h3>
  <h2
    className="animate__animated animate__backInDown"
  >
    Cargando...
  </h2>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas con el componente <GifGrid/> debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3>
     
    samurai
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="cualquier cosa"
      url="https://localhost.jpg"
    />
  </div>
</Fragment>
`;
